i6_models/parts/frontend/vgg_act.py

```diff
@@ -0,0 +1,260 @@
+from __future__ import annotations
+
+__all__ = [
+    "VGG4LayerActFrontendV1",
+    "VGG4LayerActFrontendV1Config",
+]
+
+from dataclasses import dataclass
+from typing import Callable, Optional, Tuple, Union
+
+import torch
+from torch import nn
+
+from i6_models.config import ModelConfiguration
+
+from .common import get_same_padding, mask_pool, calculate_output_dim
+
+
+@dataclass
+class VGG4LayerActFrontendV1Config(ModelConfiguration):
+    """
+    Attributes:
+        in_features: number of input features to module
+        conv1_channels: number of channels for first conv layer
+        conv2_channels: number of channels for second conv layer
+        conv3_channels: number of channels for third conv layer
+        conv4_channels: number of channels for fourth conv layer
+        conv_kernel_size: kernel size of conv layers
+        conv_padding: padding for the convolution
+        pool1_kernel_size: kernel size of first pooling layer
+        pool1_stride: stride of first pooling layer
+        pool1_padding: padding for first pooling layer
+        pool2_kernel_size: kernel size of second pooling layer
+        pool2_stride: stride of second pooling layer
+        pool2_padding: padding for second pooling layer
+        activation: activation function at the end
+        out_features: output size of the final linear layer
+    """
+
+    in_features: int
+    conv1_channels: int
+    conv2_channels: int
+    conv3_channels: int
+    conv4_channels: int
+    conv_kernel_size: Tuple
```

Comment:
You could do `Tuple[int, int]` everywhere so that it is clear to the user a two-sized tuple is expected.

Reply:
Yes, not just "could", `Tuple` should always be specified.

But then, I still would allow `Union[int, Tuple[int, int]]` here, just as PyTorch does for `Conv2d` etc.

Reply:
But then I could just revert to `IntTupleIntType`

On what should we agree? Otherwise, I'll just leave as is :)

Reply:
So what is the state here?

Reply:
Only allowing tuple makes the code more readable and allowing int and tuple is more consistent with PT. Since we are not adding any expressive by allowing int, I would leave as is. if this is something that becomes necessary in the future we can make a new PR.

Reply:
Yes I know, and I argued for allowing simply `int`, for consistency with PyTorch, and for simplicity for the user. It's very uncommon that you would pass a tuple here. So if you want simplicity, maybe just allow an `int`? But my question was, what do we do now? Was this discussed in our meeting?

Reply:
No, it was not discussed. The people present did not have strong feelings in either direction. So I decided to leave it as is.

Reply:
see meeting notes sent by Willi

Reply:
Ok then.